networkit/cpp/Unittests-X.cpp

```diff
@@ -75,13 +75,14 @@ int main(int argc, char **argv) {
 	// feenableexcept(FE_ALL_EXCEPT);
 #endif
 
-	std::string program_name = argv[0];
-	// PARSE OPTIONS
-	argc-=(argc>0); argv+=(argc>0); // skip program name argv[0] if present
+	std::string program_name(argc ? argv[0] : "");
 
-	OptionParser::Stats  stats(usage, argc, argv);
+	// PARSE OPTIONS
+	const auto parse_argc = std::max(0, argc - 1);
+	const auto parse_argv = argv + (argc>0); // skip program name argv[0] if present
```

Comment:
If I understand this change correctly, we need these local variables here as we still need the original argv for gtest, right?

Reply:
Yes, this was the underlying bug, which triggered this PR. I first generated `stats` and `option` via an lambda, to scope those vars. But this seemed to make it much harder to read.